keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,560 @@
+
+#### GA
+
+- More than one ExternalSigner implementations are completed.
+- Feature is tuned with feedback from distributions.
+
+### Upgrade/Downgrade Strategy
+
+Existing clusters preserving existing functionality: 
+- Need no changes.
+
+Existing clusters using enhanced functionality: 
+- Need a MasterVM workload that implements ExternalJWTSigner
+- Need to configure the cluster with the `--service-account-signing-endpoint` flag. 
+
+
+### Version Skew Strategy
+
+The change will completely live in the Control plane and thus, there is no question of a skew with component interaction. The only potential for a skew arises with different kube-apiserver versions running at the same time
+
+#### Different kube-apiserver versions running at the same time
+- Skew is likely to exist when externalSigning is enabled/disabled in a live cluster.
+- Distributions can take one of 2 approaches to handle it:
+  1. Tolerate disruption
+    - Disruption would exist until all kube-apiserver instances have not restarted with the new config (`--service-account-signing-endpoint` being set/unset).
+    - During this, tokens can be issued by kube-apiserver running older version and fail auth when calling kube-apiserver running newer version.
+    - The opposite is also true.
+  2. Include old keys in supported set
+    - This is for distributions that are looking for a smoother transition when enabling/disabling external singing on live clusters.
+    - When disabling, distributions shall import keys from external signers to the on-disk key set.
+    - When enabling, distributions shall implement an external signer such that it combines the external keys with on-disk keys. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+For distributions that require capability to enable the feature on a live cluster:
+- ExternalJWTSigner needs to combine the supported keys that are listest from external key management solutions and the keys that were on the cluster before enablement.
+- Control plan will need a re-start with `--service-account-key-file` being un-set and `--service-account-signing-endpoint` being set.
+- former value of `--service-account-key-file` (a path on the file system) will need to be supplied to ExternalJWTSigner in some fashion(this is up to the owners of the distribution).
+
+###### Does enabling the feature change any default behavior?
+
+Not from an end-user perspective. 
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+It will be possible but will need additional work from respective distributions. Thus, each distribution can decide for themselves.
+
+Can be achieved by satisfying the following:
+- Control plan will need a re-start with `--service-account-key-file` set and `--service-account-signing-endpoint` un-set.
```

Comment:
```suggestion
- kube-apiserver will need a restart with `--service-account-key-file` set and `--service-account-signing-endpoint` un-set.
```

Need to mention the `--service-account-signing-key-file` as well?

Reply:
Done